javascript/ql/test/library-tests/TaintTracking/object-bypass-sanitizer.js

```diff
@@ -1,10 +1,20 @@
 import * as dummy from 'dummy';
 
+function sanitizer_id(x) {
+    if (really_complicated_reason(x))
+        return x;
+    return null;
+}
+
 function f(x) {
     if (isSafe(x)) {
         sink(x);
         sink(x.foo); // NOT OK
     }
+
+    sink(sanitizer_id(x)); // OK
+    sink(sanitizer_id(x.foo)); // OK
+    sink(sanitizer_id(x).foo); // NOT OK
 }
```

Comment:
Could you duplicate `f` into two test methods? 
Such that we have one function that is invoked like: `f(source());`, and another function that is invoked like `f({foo: source()});`. 
That would make it easier to understand what is supposed to happen for each case.